### ISSUE TITLE
[CFE]: Don't register Product Collection in widgets to avoid widgets crash

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#53640]
### Which PR needs to be included? (please do not enter multiple PRs)

https://github.com/woocommerce/woocommerce/pull/53638

### Why do these PRs need to break the code freeze?

Recently, we exposed Product Collection collections to the inserter (https://github.com/woocommerce/woocommerce/pull/52209 - WooCommerce 9.5). They use a `createBlocksFromInnerBlocksTemplate` method to render which internally has additional check for blocks and it [throws an error](https://github.com/WordPress/gutenberg/blob/c45b3337753bbfe5500cbb50d38ce892b0f34a5c/packages/blocks/src/api/utils.js#L282) when the block is not registered. Many of the product elements are not registered in the Widgets area as they don't make much sense in there. Due to that, inner blocks of Product Collection block are unavailable hence the error is thrown and Widget area crashes.

Before https://github.com/woocommerce/woocommerce/pull/52209 (that will be shipped with WooCommerce 9.5), the Product Collection initially rendered the Collection Chooser, and only after the user clicked the actual block instance, `createBlocksFromInnerBlocksTemplate` got invoked and editor crashed. Now, the variations are visible in the inserter, so the actual block loads immediately and `createBlocksFromInnerBlocksTemplate` is invoked throwing the unhandled error that causes the crash.

Product Collection is not meant to be used in Widget areas and should've been hidden from the beginning.

### What is the consequence if this exception does not get included?

1. User must use classic theme
2. User enters widgets
3. Widget area crashes when any product collection variation appears in inserter search results. This happens in two cases:
    - For new stores (or those not using collections) when searching for blocks and some collection appears in results. Example: user types "o" and "On Sale Products" appear in results which result in crash (Video 1)
    - For stores that use collections a lot and they appear in inserter as defaults, it crashes instantly when opening (Video 2)

Blocks that causes a crash when appearing in inserter: 
- Best Sellers
- Hand-Picked Products
- New Arrivals
- On Sale Products
- Top Rated Products
- Related Products
- Featured Products
- Upsells
- Cross-sells

### Video 1:
https://github.com/user-attachments/assets/d02b4100-20eb-46f8-bbce-5e7b63ca53a0

### Video 2:
https://github.com/user-attachments/assets/eb568ffd-a2fa-419a-b76e-74a9beee25f4

### What is the plan should defects to be discovered in these PRs ahead of final release?

Contacting @woocommerce/woo-fse

### How should this change be communicated in the release post on the public developer blog:

No need to communicate (regression wasn't exposed publicly).

### If you’re not available and we have questions about this request, is there another person(s) and/or a team that we can ping?

@gigitux 

### Acknowledgement

- [x] I understand that I need to write an incident report (aka Escape Analysis) as a comment on this post. This is required for the request to be accepted.
- [x] I understand that I need to create an issue as a result of the Escape Analysis and reference it in a comment on this post. This can be done at a later time, but it is required for this request to be closed.
- [x] I understand that I will need to merge the PR to the target `release` branch for which the fix was intended, when my request is approved
- [x] I understand that I will need to merge the followup cherry-picked PR created by the github-actions bot to `trunk` when my request is approved

### Closing checklist for Solaris

- [ ] Escape analysis completed
- [x] The fix is raised from intended release branch. For a fix targeting WC X.Y, the base branch should be `release/X.Y`
- [x] PR merged to target `release` branch
- [ ] PR cherry-picked to `trunk` (you can find the links in PR comments to verify)
- [ ] Followup issue if any

**Comment by gigitux** [comment]
## Escape Analysis

Currently, all Woo blocks in the Widget Area are registered by default with few exceptions: https://github.com/woocommerce/woocommerce/blob/48eb0030dce131947367d632b8dfecedb2d71def/plugins/woocommerce/src/Blocks/BlockTypesController.php#L473-L489

However, coverage for the Widget Area is typically not added. To address this, [a discussion has been initiated](https://github.com/woocommerce/woocommerce/discussions/53659#discussion-7663631) to change the registration of blocks in the Widget Area from an opt-out mechanism to an opt-in one.
